tfjs-core/src/tensor.ts

```diff
@@ -353,6 +365,35 @@ export class Tensor<R extends Rank = Rank> {
     return data as Promise<DataTypeMap[D]>;
   }
 
+  /**
+   * Copy the tensor's data to a new GPU resource. Comparing to the `dataSync()`
+   * and `data()`, this method prevents data from being downloaded to CPU.
+   *
+   * For WebGL backend, the data will be stored on a densely packed texture.
+   * This means that the texture will use the RGBA channels to store value.
+   *
+   * @param options:
+   *     For WebGL,
+   *         - customTexShape: Optional. If set, will use the user defined
+   *     texture shape to create the texture.
+   *
+   * @returns For WebGL backend, a GPUResource contains the new texture and
+   *     its information.
+   *     {
+   *        dataId: Object,
+   *        texture: WebGLTexture,
+   *        texShape: [number, number] // [height, width]
+   *     }
+   *     Remember to dispose the GPUResource after it is used by
+   *     `tf.dispose(GPUResource as {} as Tensor)`.
+   *
+   * @doc {heading: 'Tensors', subheading: 'Classes'}
+   */
+  dataToGPU(options?: DataToGPUOptions): GPUResource {
```

Comment:
Just curious, why support `customTexShape`? Is there any special requirement for this?